### PR TITLE
Ubuntu devenv

### PR DESCRIPTION
**Pull request by Omarabdul3ziz**

### CODE REVIEW

File: ubuntu_devenv/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+RUN curl -sL https://deb.nodesource.com/setup_14.x  | bash - &&\
+    apt -y install nodejs &&\
+    npm install -g yarn
+
+RUN sudo apt install -y debian-keyring debian-archive-keyring apt-transport-https &&\
+    curl -1sLf 'https://dl.cloudsmith.io/public/caddy/stable/gpg.key' | sudo tee /etc/apt/trusted.gpg.d/caddy-stable.asc &&\
+    curl -1sLf 'https://dl.cloudsmith.io/public/caddy/stable/debian.deb.txt' | sudo tee /etc/apt/sources.list.d/caddy-stable.list &&\
+    sudo apt update &&\
+    sudo apt install caddy
+
```
Comment (xmonader) [inline ubuntu_devenv/Dockerfile:42]: https://github.com/coder/code-server isn't included?

File: ubuntu_devenv/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+RUN curl -LO https://dl.google.com/go/go1.17.6.linux-amd64.tar.gz &&\
+    rm -rf /usr/local/go && tar -C /usr/local -xzf go1.17.6.linux-amd64.tar.gz &&\
+    mkdir -p $HOME/go/{bin,src} 
+
+RUN git clone https://github.com/vlang/v &&\
+    cd v &&\
+    make 
+
+RUN curl -sL https://deb.nodesource.com/setup_14.x  | bash - &&\
```
Comment (xmonader) [inline ubuntu_devenv/Dockerfile:33]: update node to 16

File: ubuntu_devenv/Dockerfile (hunk 1 above)
Comment (xmonader) [reply ubuntu_devenv/Dockerfile:42]: actually found it is in the image, please upgrade the version to 4.x instead of the bundled 3.x

Comment (Omarabdul3ziz) [reply ubuntu_devenv/Dockerfile:42]: I think in the last build we load that layer `RUN curl -fsSL https://code-server.dev/install.sh | sh` from cache which created before the `code-server 4` released, `docker build --no-cache` fixes this and install `code-server v4.0.1`

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**